provider/zone_id_filter.go

```diff
@@ -0,0 +1,47 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package provider
+
+import (
+	"github.com/aws/aws-sdk-go/service/route53"
+)
+
+// ZoneIDFilter holds a list of zone ids to filter by
+type ZoneIDFilter struct {
+	zoneIDs []string
+}
+
+// NewZoneIDFilter returns a new ZoneIDFilter given a list of zone ids
+func NewZoneIDFilter(zoneIDs []string) ZoneIDFilter {
+	return ZoneIDFilter{zoneIDs}
+}
+
+// Match checks whether a zone matches one of the provided zone ids
+func (f ZoneIDFilter) Match(zone *route53.HostedZone) bool {
```

Comment:
if we introduce `--aws-zone-id` and zone filtering only for AWS, then either `ZoneIDFilter` should be moved to `aws.go` file, or this filter renamed to `aws_zone_id_filter.go` or (which I would prefer) `Match` method should accept a plain string (`zoneID`) so that we can reuse it for other providers